pkgs/tools/typesetting/pulldown-cmark/default.nix

```diff
@@ -0,0 +1,29 @@
+{ lib
+, rustPlatform
+, fetchFromGitHub
+}:
+
+let
+  owner = "raphlinus";
+  pname = "pulldown-cmark";
+  version = "0.9.2";
+in
+rustPlatform.buildRustPackage {
+  inherit pname version;
+
+  src = fetchFromGitHub {
+    inherit owner;
+    repo = pname;
+    rev = "v${version}";
+    hash = "sha256-AAb+dSJ1oSRuvWu47VvzCeB6pQE6/+u69io2FsZoZHM=";
+  };
```

Comment:
```suggestion
rustPlatform.buildRustPackage rec {
  pname = "pulldown-cmark";
  version = "0.9.2";

  src = fetchFromGitHub {
    owner = "raphlinus";
    repo = pname;
    rev = "v${version}";
    hash = "sha256-AAb+dSJ1oSRuvWu47VvzCeB6pQE6/+u69io2FsZoZHM=";
  };
```

Reply:
`rec` is preferred over `let ... in`? I assumed `rec` was to be avoided to prevent accidental infinite recursion.

Reply:
I was thinking if I remove the owner variable I might as well condense the other part a bit. rec vs let in does not really matter here and the infinite recursion part applies more to bigger files and functions.